src/App.js

```diff
@@ -0,0 +1,254 @@
+  const [status, setStatus] = useState('');
+
+  const changeHandler = (event) => {
+    setSelectedFile(event.target.files[0]);
+		setIsFilePicked(true);
+    Papa.parse(event.target.files[0], {
+      header: true,
+      skipEmptyLines: true,
+      complete: function (results) {
+        setParsedData(results.data);
+        setNewData(results.data);
+      },
+    });
+
+  }
+  
+  const handleSubmission = () => {
+    setActiveData(0);
+    handleData(0);
+	}
+
+  const handleModalClose = () => {
+    setActiveData('');
+    setIsFilePicked(false);
+    handleDownload();
+  }
+
+  const handleNextData = () => {
+    const length = parsedData.length;
+    var index = activeData + 1;
+    index = index % length;
+    setActiveData(index);
+    handleData(index);
+  }
+
+  const handlePreviousData = () => {
+    const length = parsedData.length;
+    var index = activeData;
+    index = index === 0 ? length - 1 : index - 1;
+    setActiveData(index);
+    handleData(index);
+  }
+
+  function handleData(index) {
+    setOldUrl(parsedData[index].oldUrl);
+    setNewUrl(parsedData[index].newUrl);
+    setStatus(newData[index].status);
+  }
+
+  const handleAccept = () => {
+    setNewData(prevNewData => prevNewData.map((data, i) => {
+      if (i === activeData) {
+        return {
+          ...data,
+          status: 'PASSED'
+        }
+      }
+      return data;
+    }));
+    handleNextData();
+  }
+
+  const handleReject = () => {
+    setNewData(prevNewData => prevNewData.map((data, i) => {
+      if (i === activeData) {
+        return {
+          ...data,
+          status: 'FAILED'
+        }
+      }
+      return data;
+    }));
+    handleNextData();
+  }
+
+  const handleDownload = () => {
+    const finalData = Papa.unparse(newData);
+    const csvContent = `data:text/csv;charset=utf-8,${finalData}`;
+    const encodedURI = encodeURI(csvContent);
+    window.open(encodedURI);
+  }
+
+  const handleSkip = () => {
+    let index = newData.findIndex(data => data.status === '' || data.status === undefined) ;
+    if (index > 0) {
+      setActiveData(index);
+      handleData(index);
+    }
+  }
+
+  return (
+    <div id="main-app">
+      <Header 
+        title="Image Comparison Tool" 
+      />
+      { activeData === '' ? (
+        <div className='image-comparison-tool'>
+          <Instructions />
+          <input type="file" name="file" className="input-file" onChange={changeHandler} accept=".csv" />
+          { isFilePicked ? (
+            <FileUpload  
+              fileName={selectedFile.name}
+              fileType={selectedFile.type}
+              fileSize={selectedFile.size}
+              fileDate={selectedFile.lastModifiedDate.toLocaleDateString()}
+            />
+          ) : (
+            <div>
+              <p>Select a file to show details</p>  
+            </div>
+          )}
+          <button onClick={handleSubmission}>Start comparing images</button>
+        </div>
```

Comment:
**Suggestion (non-blocking)**: Totally optional, but I've sometimes found that combining inline ternary expressions with bigger blocks of elements can be a little bit harder to read. I recommend wrapping these into their own bigger component so that it's more easy to see what is exactly changing in the ternary expression.

Reply:
> I recommend wrapping these into their own bigger component so that it's more easy to see what is exactly changing in the ternary expression.

I agree with this! I have moved them into their own component so it will be easier to read. 

**Thank you so much for taking the time to review the initial changes 🎉**